plugin/keyring_vault/vault_curl.cc

```diff
@@ -0,0 +1,299 @@
+#include <my_global.h>
+#include <boost/scope_exit.hpp>
+#include <algorithm>
+#include "my_rdtsc.h"
+#include "sql_error.h"
+#include "mysqld.h"
+#include "mysql/service_thd_wait.h"
+#include "vault_curl.h"
+#include "vault_memory.h"
+#include "vault_base64.h"
+
+namespace keyring
+{
+
+static const size_t max_response_size = 32000000;
+static const long timeout = 300; //5m timeout
+static MY_TIMER_INFO curl_timer_info;
+static ulonglong last_ping_time;
+static volatile bool was_thd_wait_started = false;
+static const ulonglong slow_connection_threshold = 100;//[ms]
+
+static size_t write_response_memory(void *contents, size_t size, size_t nmemb, void *userp)
+{
+  size_t realsize = size * nmemb;
```

Comment:
According to the doc this value should not be greater than CURL_MAX_WRITE_SIZE (16k). However, I have added overflow check, just to be sure.